tests/README.md

```diff
@@ -7,3 +7,155 @@ Please add following dependencies to your PATH before running `make test-integra
 * [bindle-server](https://github.com/deislabs/bindle)
 * [nomad](https://github.com/hashicorp/nomad)
 * [Hippo.Web](https://github.com/deislabs/hippo)
+
+# E2E tests for spin
+
+The goal of these tests is to ensure that spin continues to work with existing apps/examples/templates and there are no regressions introduced as we add more functionality or refactor existing code.
+
+These tests work with following deployments models:
+
+- spin up
+- Fermyon Cloud
+- Local platform (WIP)
+
+## Important files and their function
+
+`crates/e2e-testing`     - All the test framework/utilities that are required for `e2e-tests`
+`tests/testcases/mod.rs` - All the testcase definitions should be added here.
+`tests/spinup_tests.rs`  - All tests that we want to run with `Spin Up` should be added here
+`tests/cloud_tests.rs`   - All tests that we want to run with `Fermyon Cloud` should be added here
+`tests/testcases/<dirs>` - The testcases which require corresponding `spin app` pre-created should be added here
+
+
```

Comment:
Would you mind adding a `How to run e2e tests` section here?

Reply:
sure thing. updated  the doc, thanks Michelle.